apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -45,9 +44,10 @@ internal ClientPool(ISpannerClientFactory clientFactory = null)
 
         public async Task<SpannerClient> AcquireClientAsync(
             ITokenAccess credentials = null,
-            ServiceEndpoint endpoint = null)
+            ServiceEndpoint endpoint = null,
+            IDictionary additionalOptions = null)
```

Comment:
Does this need to be non-generic? Is that following a convention from somewhere? I now understand it's so that we can pass the connection string builder directly, but that means there will be all the existing aspects like the service name in there as well. Can we make it genuinely *additional* options, and restrict it to a `Dictionary<string, string>`?`